server/rest/src/main/java/org/infinispan/rest/resources/ContainerResource.java

```diff
@@ -533,13 +545,30 @@ private CompletionStage<RestResponse> shutdown(RestRequest restRequest) {
    }
 
    private CompletionStage<RestResponse> listenConfig(RestRequest request) {
+      return streamConfigurationAndEvents(request, false);
+   }
+
+   private CompletionStage<RestResponse> streamListenerEvents(RestRequest request) {
+      return streamConfigurationAndEvents(request, true);
+   }
+
+   private CompletionStage<RestResponse> streamConfigurationAndEvents(RestRequest request, final boolean includeEvents) {
       MediaType mediaType = negotiateMediaType(request, APPLICATION_YAML, APPLICATION_JSON, APPLICATION_XML);
       boolean includeCurrentState = Boolean.parseBoolean(request.getParameter("includeCurrentState"));
       EmbeddedCacheManager cacheManager = invocationHelper.getRestCacheManager().getInstance();
       NettyRestResponse.Builder responseBuilder = new NettyRestResponse.Builder();
       ConfigurationListener listener = new ConfigurationListener(cacheManager, mediaType, includeCurrentState);
       responseBuilder.contentType(TEXT_EVENT_STREAM).entity(listener.getEventStream());
-      return cacheManager.addListenerAsync(listener).thenApply(v -> responseBuilder.build());
+
+      final CompletionStage<Void> cs = cacheManager.addListenerAsync(listener);
+      if (includeEvents) {
+         Cache<?, ?> cache = cacheManager.getCache(SERVER_LOGGER_CACHE_NAME, false);
```

Comment:
Instead of retrieving the logger cache like this (which leaks an implementation detail of the server logger), we should instead add
a `EventLogger getEventLogger();` method to `ServerManagement`.
`EventLogger` should have a way to publish events to some kind of `EventLogListener`.

Reply:
I've created https://issues.redhat.com/browse/ISPN-13880 to follow-up